src/battle/other_battle_calculators.c

```diff
@@ -3114,3 +3158,270 @@ u32 RollMetronomeMove(struct BattleSystem *bsys)
 {
     return (BattleRand(bsys) % NUM_OF_MOVES) + 1;
 }
+
+// fuck it be a coward and pre-calulate the values
+const int getShakeChancesLookupTable[] = {
```

Comment:
this was the fractional exponent thing right?  should just make it into `u16` and map 255 to 65536 manually in order to cut down on a lot of space